### ISSUE TITLE
[BUG] Unable to stop on any exception #965 come back

### ISSUE THREAD

**Issue by jeanpierrep1972** [microsoft/vscode-dotnettools#1242]
### Describe the Issue

Describe the Issue
With MAUI I am unable to debug and ask VS Code to stop where there is an exception.

I use VS Code on Mac

![image](https://github.com/microsoft/vscode-dotnettools/assets/1383636/0800b333-c9dd-4b64-8576-4e7d2f7034ae)

### Expected Behavior

We should be able to set if we want to stop on all exception or unandled exception. The checkboxes are missing inside breakpoint panel

### Environment Information

I have the .NET SDK 8.0.302

With the .NET MAUI

	<PackageReference Include="Microsoft.Maui.Controls" Version="8.0.60" />
	<PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="8.0.60" />

And the extensions:

Version: 1.90.2
Commit: 5437499feb04f7a586f677b155b039bc2b3669eb
Date: 2024-06-18T22:37:41.291Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin x64 23.2.0

alefragnani.project-manager@12.8.0
aliasadidev.nugetpackagemanagergui@2.1.0
bbenoist.shell@0.3.0
cheshirekow.cmake-format@0.6.11
codezombiech.gitignore@0.9.0
dae.vscode-mac-color-picker@1.1.0
davidanson.vscode-markdownlint@0.55.0
davidmarek.jsonpath-extract@1.2.3
donjayamanne.git-extension-pack@0.1.3
donjayamanne.githistory@0.6.20
dotjoshjohnson.xml@2.5.1
eamodio.gitlens@15.1.0
editorconfig.editorconfig@0.16.4
github.remotehub@0.62.0
go2sh.cmake-integration-vscode@0.7.1
ionide.ionide-fsharp@7.20.3
jmrog.vscode-nuget-package-manager@1.1.6
jorgeserrano.vscode-csharp-snippets@1.1.0
josetr.cmake-language-support-vscode@0.0.9
k--kato.docomment@1.0.0
kaysonwu.cpptask@0.0.1
kreativ-software.csharpextensions@1.7.3
leavesster.jsonpath@0.0.7
moshfeu.compare-folders@0.24.3
ms-azure-devops.azure-pipelines@1.228.0
ms-azuretools.vscode-azureresourcegroups@0.9.1
ms-azuretools.vscode-docker@1.29.1
ms-dotnettools.csdevkit@1.7.27
ms-dotnettools.csharp@2.34.12
ms-dotnettools.dotnet-interactive-vscode@1.0.5229040
ms-dotnettools.dotnet-maui@1.1.8
ms-dotnettools.vscode-dotnet-pack@1.0.13
ms-dotnettools.vscode-dotnet-runtime@2.0.6
ms-dotnettools.vscode-dotnet-sdk@0.8.0
ms-dotnettools.vscodeintellicode-csharp@2.1.11
ms-sarifvscode.sarif-viewer@3.4.4
ms-toolsai.jupyter@2024.5.0
ms-toolsai.jupyter-keymap@1.1.2
ms-toolsai.jupyter-renderers@1.0.18
ms-toolsai.vscode-jupyter-cell-tags@0.1.9
ms-toolsai.vscode-jupyter-slideshow@0.1.6
ms-vscode-deploy-azure.azure-deploy@1.2.4
ms-vscode-remote.remote-containers@0.369.0
ms-vscode.azure-account@0.12.0
ms-vscode.azurecli@0.6.0
ms-vscode.cmake-tools@1.18.42
ms-vscode.cpptools@1.20.5
ms-vscode.hexeditor@1.10.0
ms-vscode.mono-debug@0.16.3
ms-vscode.powershell@2024.2.2
ms-vscode.remote-repositories@0.40.0
pflannery.vscode-versionlens@1.14.2
twxs.cmake@0.0.17
visualstudioexptteam.intellicode-api-usage-examples@0.2.8
visualstudioexptteam.vscodeintellicode@1.3.1
vmware.vscode-concourse@1.55.0
vscode-icons-team.vscode-icons@12.8.0
ziyasal.vscode-open-in-github@1.3.6

**Comment by jeanpierrep1972** [comment]
I created this new issue.. The bug has come back..

https://github.com/microsoft/vscode-dotnettools/issues/1242

**Comment by mauroa** [comment]
@jeanpierrep1972 it seems you are having a dotnet SDK version not compatible with the new debug engine that .NET MAUI wants to use.
This is a new feature added some time ago, where .NET MAUI will try to use the same debug engine as other C# projects (vsdbg) if you have .NET 8.0.300 or higher.
It looks like in this case, your .NET MAUI and VS Code is using the old Mono engine, which doesn't support configuring exceptions for debugging.

There's also a chance that for some reason, you have that setting disabled. Please check in the VS Code settings, under the .NET MAUI section, for a Use VSDbg setting, and ensure it's enabled. Also, please ensure that you have a compatible .NET version.

You can share the content of the C#, C# DevKit and .NET MAUI output panes in case it still doesn't work.
Thanks

**Comment by jeanpierrep1972** [comment]
When I put bacj this settings to ON, I can see the Exceptions checkboxes in the debugger panes:

![image](https://github.com/microsoft/vscode-dotnettools/assets/1383636/35f08707-d89b-4b31-bd25-ad8622f95f91)

But with this option ON it's really hard to debug an application... I'm able to stop at a breakpoint, but as soon as I do Step Over, the debugger and the iOS application in the simulator freeze.

**Comment by dingxiaolong** [comment]
Just like the comment say above ,   the debug will lose connect when try to step over ,  step in  ,  I can't even do a simple debug for all my projects with Vs code.    
I raise a ticket before ,  but there is no response or plan when this can be fixed or will be fixed ,   this is a very basic and very important feature .  it should be put the priority list .

**Comment by jeanpierrep1972** [comment]
I agree that the debugging functionality should be a priority because VS Studio Mac is decommissioned and VS Studio, which is supposed to replace it, doesn't do the job 100% for a task as simple as Step Over, Step Into etc.

**Comment by mauroa** [comment]
I'm closing this because the lack of activity in more than a year and because the new Mono debug engine shipped in the C# extension has received many fixes and changes during this time, so the reported issues should be fixed. This engine is now enforced as the default one, so you won't see the VS Code setting anymore.
Please ensure to update to the latest C#, C# DevKit and .NET MAUI versions.
Thanks
